zerver/lib/events.py

```diff
@@ -1251,7 +1254,11 @@ def _draft_update_action(i: int) -> None:
             for stream in event["streams"]:
                 stream_data = copy.deepcopy(stream)
                 if include_subscribers:
-                    stream_data["subscribers"] = []
+                    stream_data["subscribers"] = list(
+                        get_active_subscriptions_for_stream_id(
+                            stream["stream_id"], include_deactivated_users=False
+                        ).values_list("user_profile_id", flat=True)
+                    )
```

Comment:
Hmm, this seems like an odd pattern. I'd have expected either the subscribers to already be present in the create event, or to be sent via a separate `peer_add` event. 

We also generally aim for no database queries in this code path, though it is OK to have occasional exceptions.

Reply:
I think this code is probably correct, despite not being how we usually want to do things.

But I think it does reflect an actual issue. Clients will not receive subscriber data for the channel at all if they receive the event, rather than hitting this rare race. So this needs an "api design" follow-up conversation.

Reply:
OK, I think we don't need an API design discussion, in that there's a pretty clear precedent for determining what to do here -- the case of getting added to a private channel and gaining metadata access to it that way. And that currently works via sending `peer_add` events, as discussed in https://github.com/zulip/zulip/pull/33896/commits/24639118d490b5c9a43c2cd5b3cb4445b987e2fb#r1993285770